compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/ProjectLoader.java

```diff
@@ -71,7 +71,7 @@ public static Project loadProject(Path path, ProjectEnvironmentBuilder projectEn
             return BuildProject.load(projectEnvironmentBuilder, projectRoot, buildOptions);
         }
 
-        if (!ProjectPaths.isBalFile(absFilePath)) {
+        if (!ProjectPaths.isBalFile(absFilePath) && !ProjectPaths.isBallerinaRelatedToml(absFilePath)) {
```

Comment:
We are addressing the necessary changes in utilities to load a Bala project in #28857. IMO from the LS, we only need to switch to use the `ProjectLoader.loadProject` instead of the `BuildProject.load` and `SingleFileProject.load` methods.

Reply:
Okay, I'll revert this change and this PR should be merged after #28857.